### PR TITLE
Add media playback badge for Area card

### LINKED ISSUE #26684: Media playback badge for Area card
### Description

Adding button controls for media players is very complicated UX wise but we can add badge if there is a media player in the area with state "playing". Similar to motion/occupancy.

Image

### Additional context

From #27413
- deprecate alert_classes
- introduce a new option badges that can contains more than alert (binary_sensor, media and even lights in the future)

### PR DESCRIPTION
**Pull request by winterborn**
## Proposed change

This PR implements issue #26684 by adding a media playback badge to Area cards, similar to existing motion/occupancy badges. The badge shows when media players in an area are playing or paused, providing quick visual feedback about active media in each room.

**Key Features:**
- Shows play/pause icon badge when media players are active in an area
- Displays appropriate label ("Media playing" or "Media paused")
- Only appears in compact mode, following existing badge patterns
- Respects area boundaries (only shows media players assigned to that specific area)

image

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
# No configuration needed - works automatically with existing Area cards
# Just ensure media players are assigned to areas in Settings > Areas & Zones
```

## Additional information

- This PR fixes or closes issue: fixes #26684
- This PR is related to issue or discussion: https://github.com/home-assistant/frontend/issues/26684
- Link to documentation pull request:

**Files Changed:**
- `src/common/entity/compute_active_media_states.ts` (new utility function)
- `src/panels/lovelace/cards/hui-area-card.ts` (badge integration)
- `test/common/entity/compute_active_media_states.test.ts` (test suite)

**Testing:**
- Added test cases covering happy path and edge cases
- Manually tested with media players in different states
- Verified badge appears/disappears based on media player state changes
- Confirmed area boundary respect (only shows for assigned media players)

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/common/entity/compute_active_media_states.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import type { HassEntity } from "home-assistant-js-websocket";
+import type { HomeAssistant } from "../../types";
+import { generateEntityFilter } from "./entity_filter";
+
+export function computeActiveMediaStates(
```
Comment (bramkragten) [inline src/common/entity/compute_active_media_states.ts]: Can we move this to `/data/media-player.ts` to keep all media player related logic in one place?

File: src/common/entity/compute_active_media_states.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  hass: HomeAssistant,
+  areaId?: string
+): HassEntity[] {
+  const area = areaId ? hass.areas[areaId] : undefined;
+  if (!area) {
+    return [];
```
Comment (bramkragten) [inline src/common/entity/compute_active_media_states.ts]: `areaId` is optional, and the function name doesn't imply that this function will only work for areas, and yet if you don't pass an `areaId` the function returns an empty array?

Make the name of the function more descriptive, like get `computeActiveAreaMediaStates`

File: src/panels/lovelace/cards/hui-area-card.ts [hunk 3]
```diff
@@ -302,6 +307,27 @@ export class HuiAreaCard extends LitElement implements LovelaceCard {
...
   }
 
+  private _renderMediaBadge(): TemplateResult<1> | typeof nothing {
+    const states = this._computeActiveMediaStates();
+
+    if (states.length === 0) {
+      return nothing;
+    }
+
+    const firstState = states[0];
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-area-card.ts]: Why do we use the first state? Shouldn't `playing` take precedent over `paused` for example?

File: src/common/entity/compute_active_media_states.ts (hunk 1 above)
Comment (winterborn) [reply src/common/entity/compute_active_media_states.ts]: Moved the helper to `/src/data/media-player.ts` to keep media-player logic together. Updated imports.
Commit: [56854be](https://github.com/home-assistant/frontend/pull/26893/commits/56854be2b759da16f942ee17e52bb19271e4485c)

File: src/common/entity/compute_active_media_states.ts (hunk 2 above)
Comment (winterborn) [reply src/common/entity/compute_active_media_states.ts]: Renamed the helper to `computeActiveAreaMediaStates` and made `areaId` required so the contract matches the name/usage. Updated call sites.
Commit: [56854be](https://github.com/home-assistant/frontend/pull/26893/commits/56854be2b759da16f942ee17e52bb19271e4485c)

File: src/panels/lovelace/cards/hui-area-card.ts (hunk 3 above)
Comment (winterborn) [reply src/panels/lovelace/cards/hui-area-card.ts]: Replaced first state logic with precedence: `playing` > `paused`, otherwise no badge.
Commit: [56854be](https://github.com/home-assistant/frontend/pull/26893/commits/56854be2b759da16f942ee17e52bb19271e4485c)

**Review by silamon — COMMENTED** [review]
It works like it should be, unfortunately we don't have a good way to display multiple tile badges.

**Comment by winterborn** [comment]
> Yeah, those design processes are quite slow unfortunately. I think that we can solve easily without UX if you ensure that the alert tile badge is display with a higher priority than the music playing/paused.

Added [here](https://github.com/home-assistant/frontend/pull/26893/commits/72f1a1cc8ba24b62ba14f9644fbe094d063a2a31)

**Comment by bramkragten** [comment]
I think we should not show paused media players, like Sonos media players will always be paused if they are not playing. We should only show it when it is playing media.

File: src/data/media-player.ts [hunk 4]
```diff
@@ -520,3 +522,33 @@ export const mediaPlayerJoin = (
...
+/**
+ * Compute active media player states in a specific area.
+ * @param hass Home Assistant object
+ * @param areaId Area ID to filter media players by
+ * @returns Array of active media player entities (playing or paused)
+ */
+export const computeActiveAreaMediaStates = (
+  hass: HomeAssistant,
+  areaId: string
+): HassEntityBase[] => {
```
Comment (bramkragten) [inline src/data/media-player.ts]:
```suggestion
): MediaPlayerEntity[] => {
```

File: src/data/media-player.ts [hunk 5]
```diff
@@ -520,3 +522,33 @@ export const mediaPlayerJoin = (
 
 export const mediaPlayerUnjoin = (hass: HomeAssistant, entity_id: string) =>
   hass.callService("media_player", "unjoin", {}, { entity_id });
+
+/**
+ * Compute active media player states in a specific area.
+ * @param hass Home Assistant object
+ * @param areaId Area ID to filter media players by
+ * @returns Array of active media player entities (playing or paused)
```
Comment (bramkragten) [inline src/data/media-player.ts]:
```suggestion
 * @returns Array of playing media player entities
```

File: src/panels/lovelace/cards/hui-area-card.ts [hunk 6]
```diff
@@ -563,7 +585,11 @@ export class HuiAreaCard extends LitElement implements LovelaceCard {
           <div class="content ${classMap(contentClasses)}">
             <ha-tile-icon>
               ${displayType === "compact"
-                ? this._renderAlertSensorBadge()
+                ? html`
+                    ${this._computeActiveAlertStates().length > 0
+                      ? this._renderAlertSensorBadge()
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-area-card.ts]: We basically call `_computeActiveAlertStates` twice here, can we store the result or memoize the function?

File: src/translations/en.json [hunk 7]
```diff
@@ -103,7 +103,9 @@
         }
       },
       "area": {
-        "area_not_found": "Area not found."
+        "area_not_found": "Area not found.",
+        "media_playing": "Media playing",
+        "media_paused": "Media paused"
```
Comment (bramkragten) [inline src/translations/en.json]:
```suggestion
```

File: src/data/media-player.ts (hunk 4 above)
Comment (winterborn) [reply src/data/media-player.ts]: Changes [here](https://github.com/home-assistant/frontend/pull/26893/commits)

File: src/data/media-player.ts (hunk 5 above)
Comment (winterborn) [reply src/data/media-player.ts]: Changes [here](https://github.com/home-assistant/frontend/pull/26893/commits)

File: src/panels/lovelace/cards/hui-area-card.ts (hunk 6 above)
Comment (winterborn) [reply src/panels/lovelace/cards/hui-area-card.ts]: Changes [here](https://github.com/home-assistant/frontend/pull/26893/commits)

File: src/translations/en.json (hunk 7 above)
Comment (winterborn) [reply src/translations/en.json]: Changes [here](https://github.com/home-assistant/frontend/pull/26893/commits)

**Review by bramkragten — APPROVED** [review]

**Comment by piitaya** [comment]
Sorry, we had to revert the PR. After discussion with the rest of the team, it should implemented that way (more info in the 27413 PR)
